src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorFeatureDetector.cs

```diff
@@ -28,6 +28,7 @@ public DefaultLSPEditorFeatureDetector(ProjectHierarchyInspector projectHierarch
             {
                 throw new ArgumentNullException(nameof(projectHierarchyInspector));
             }
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
To do foreground thread checks like this you'll need to use `_joinableTaskContext.IsOnMainThread` to keep these methods unit testable.